test/parallel/test-dgram-socket-buffer-size.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const dgram = require('dgram');
+
+{
+  // Should throw error if the socket is never bound.
+  const socket = dgram.createSocket('udp4');
+
+  assert.throws(() => {
+    socket.setRecvBufferSize(8192);
+  }, common.expectsError({
+    code: 'ERR_SOCKET_BUFFER_SIZE',
+    type: Error,
+    message: /^Could not get or set buffer size:.*$/
+  }));
+
+  assert.throws(() => {
+    socket.setSendBufferSize(8192);
+  }, common.expectsError({
+    code: 'ERR_SOCKET_BUFFER_SIZE',
+    type: Error,
+    message: /^Could not get or set buffer size:.*$/
+  }));
+
+  assert.throws(() => {
+    socket.getRecvBufferSize();
+  }, common.expectsError({
+    code: 'ERR_SOCKET_BUFFER_SIZE',
+    type: Error,
+    message: /^Could not get or set buffer size:.*$/
+  }));
+
+  assert.throws(() => {
+    socket.getSendBufferSize();
+  }, common.expectsError({
+    code: 'ERR_SOCKET_BUFFER_SIZE',
+    type: Error,
+    message: /^Could not get or set buffer size:.*$/
+  }));
+}
+
+{
+  // Should throw error if invalid buffer size is specified
+  const socket = dgram.createSocket('udp4');
+
+  socket.bind(common.mustCall(() => {
+    assert.throws(() => {
+      socket.setRecvBufferSize(-1);
+    }, common.expectsError({
+      code: 'ERR_SOCKET_BAD_BUFFER_SIZE',
+      type: Error,
```

Comment:
Yes it should, I am throwing TypeError in dgram.js, however the test is passing. Regardless I will update the test.